src/app/course/goal-creation-page/goal-creation-page.component.ts

```diff
@@ -0,0 +1,166 @@
+import {Component, OnInit} from '@angular/core'
+import {FormArray, FormControl, FormGroup} from "@angular/forms"
+import {GoalForm} from "@app/course/_forms/goal.form"
+import {GoalService} from "@app/course/_services/goal.service"
+import {ActivatedRoute, Router} from "@angular/router"
+import {TuiNotification, TuiNotificationsService} from "@taiga-ui/core"
+import {tuiCreateTimePeriods} from "@taiga-ui/kit"
+import {CategoryService} from "@app/_services/api/category.service"
+import {Category} from "@app/_models"
+import {DifficultyService} from "@app/problems/_services/difficulty.service"
+import {Difficulty} from "@app/_models/difficulty"
+import {Goal, GoalItem} from "@app/_models/goal/goal"
+
+@Component({
+    selector: 'app-goal-creation-page',
+    templateUrl: './goal-creation-page.component.html',
+    styleUrls: ['./goal-creation-page.component.scss']
+})
+export class GoalCreationPageComponent implements OnInit {
+    constructor(
+        private readonly goalService: GoalService,
+        private readonly categoryService: CategoryService,
+        private readonly difficultyService: DifficultyService,
+        private readonly router: Router,
+        private readonly activatedRoute: ActivatedRoute,
+        private readonly notificationService: TuiNotificationsService
+    ) {
+    }
+
+    timeOptions = tuiCreateTimePeriods()
+    goalForm: FormGroup
+    categories: Category[]
+    difficulties: Difficulty[]
+    courseId: number
+
+    public isLoaded = false
+    suggestedGoals: Goal[]
+    suggestedGoalsStr: [string[]] = [[]]
+
+    ngOnInit(): void {
+        this.courseId = this.activatedRoute.snapshot.params.courseId
+        this.goalForm = GoalForm.createGoalForm()
+        this.categoryService.getCategories().subscribe(data => {
+            this.categories = data
+
+            this.goalService.getGoalSuggestions().subscribe(data => {
```

Comment:
this does not need to be inside the subscribe. It it causing unnecessary nested subscriptions Please move it out.